backends/zypp/pk-backend-zypp.cpp

```diff
@@ -154,10 +156,46 @@ guint _dl_count = 0;
 guint _dl_progress = 0;
 guint _dl_status = 0;
 
+
+static gboolean
+zypp_refresh_repo_appdata (RepoInfo &repo)
+{
+	Pathname metadataPath = repo.metadataPath();
+	std::string metadataPathString = metadataPath.c_str();
+	xml::Reader reader(metadataPathString + "/repomd.xml");
+
+	while (!reader.atEnd())
+	{
+		reader.nextNode();
+		std::string location = reader->getAttribute("href").c_str();
+		if (location.find("appdata.xml.gz") != std::string::npos ||
+		    location.find("appdata-icons.xml.gz") != std::string::npos ||
+		    location.find("primary.xml.gz") != std::string::npos)
```

Comment:
Doesn't this logic mean that it will also attempt this if `primary.xml.gz` exists but not the other two? That obviously would be quite broken...

Reply:
I didn't notice any errors in testing with this logic. The repositories with appdata cached theirs just fine and the other repositories didn't cause any errors.

Reply:
I would suggest going for that anyway, get the hrefs into an array and iterate over it to install, checking if all the required files are there.

We also probably should change the names of those files in our repos so the backend works with Fedora and other dnf based distros, as well as dnf work on openSUSE and other zypp based distros.